RecoJets/JetProducers/plugins/FixedGridRhoProducerFastjetFromRecHit.cc

```diff
@@ -0,0 +1,180 @@
+#include "fastjet/tools/GridMedianBackgroundEstimator.hh"
+#include "FWCore/Framework/interface/Event.h"
+#include "DataFormats/Common/interface/View.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "TLorentzVector.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "RecoEgamma/EgammaIsolationAlgos/interface/EgammaHcalIsolation.h"
+#include "CondFormats/EcalObjects/interface/EcalPFRecHitThresholds.h"
+#include "CondFormats/DataRecord/interface/EcalPFRecHitThresholdsRcd.h"
+
+class FixedGridRhoProducerFastjetFromRecHit : public edm::stream::EDProducer<> {
+public:
+  explicit FixedGridRhoProducerFastjetFromRecHit(const edm::ParameterSet &iConfig);
+  ~FixedGridRhoProducerFastjetFromRecHit() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  void produce(edm::Event &, const edm::EventSetup &) override;
+  void getHitP4(const DetId &detId, float hitE, TLorentzVector &hitp4, const CaloGeometry &caloGeometry);
+  bool passedHcalNoiseCut(const HBHERecHit &hit);
+  bool passedEcalNoiseCut(const EcalRecHit &hit, const EcalPFRecHitThresholds *thresholds);
+
+  fastjet::GridMedianBackgroundEstimator bge_;
+  edm::EDGetTokenT<HBHERecHitCollection> hbheRecHitsTag_;
+  edm::EDGetTokenT<EcalRecHitCollection> ebRecHitsTag_;
+  edm::EDGetTokenT<EcalRecHitCollection> eeRecHitsTag_;
+
+  const EgammaHcalIsolation::arrayHB eThresHB_;
+  const EgammaHcalIsolation::arrayHE eThresHE_;
+
+  //Muon HLT currently use ECAL-only rho for ECAL isolation,
+  //and HCAL-only rho for HCAL isolation. So, this skipping option is needed.
+  bool skipHCAL_;
+  bool skipECAL_;
+
+  const edm::ESGetToken<EcalPFRecHitThresholds, EcalPFRecHitThresholdsRcd> ecalPFRechitThresholdsToken_;
+  const edm::ESGetToken<CaloGeometry, CaloGeometryRecord> caloGeometryToken_;
+};
+
+FixedGridRhoProducerFastjetFromRecHit::FixedGridRhoProducerFastjetFromRecHit(const edm::ParameterSet &iConfig)
+    : bge_(iConfig.getParameter<double>("maxRapidity"), iConfig.getParameter<double>("gridSpacing")),
+      hbheRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("hbheRecHitsTag"))),
+      ebRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("ebRecHitsTag"))),
+      eeRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("eeRecHitsTag"))),
+      eThresHB_(iConfig.getParameter<EgammaHcalIsolation::arrayHB>("eThresHB")),
+      eThresHE_(iConfig.getParameter<EgammaHcalIsolation::arrayHE>("eThresHE")),
+      skipHCAL_(iConfig.getParameter<bool>("skipHCAL")),
+      skipECAL_(iConfig.getParameter<bool>("skipECAL")),
+      ecalPFRechitThresholdsToken_{esConsumes()},
```

Comment:
```suggestion
      ecalPFRecHitThresholdsToken_{esConsumes()},
```
Consistent naming (RecHit).